pkg/nfd-master/nfd-master.go

```diff
@@ -814,7 +783,9 @@ func filterExtendedResource(name, value string, features *nfdv1alpha1.Features)
 	if ns == "" {
 		return "", fmt.Errorf("extended resource without namespace (prefix/) not allowed")
 	}
-	if ns != nfdv1alpha1.ExtendedResourceNs && !strings.HasSuffix(ns, nfdv1alpha1.ExtendedResourceSubNsSuffix) {
+
+	err := validate.ExtendedResource(name, value, features)
+	if err == validate.ErrNSNotAllowed {
```

Comment:
This logic looks b0rken 🤨 What if there's some other error, we just ignore it and continue happily(?)

We should move the check of `kubernetes.io`  and `kubernetes.io` below to the validate func